src/com/mjcode/model/Piece.java

```diff
@@ -40,55 +34,70 @@ public boolean canCapture(Piece other) {
 
     /**
      * The canMoveTo method determines if a pieces can legally move to a specified tile
-     * @param tile The specified tile the piece wants to move to
+     * @param targetTile The specified tile the piece wants to move to
      * @return True if piece can legally move to that tile, else false
      */
-    public boolean canMoveTo(Tile tile) {
+    public boolean canMoveTo(Tile targetTile,Tile[][] board) {
+
         // Local variables to store the squares that lies between the current tile and the target tile
         int jumpedRow;
         int jumpedCol;
 
-        // Calculate the row and column differences between the current tile and target tile
-        int rowDiff = tile.getRow() - this.currentRow;
-        int colDiff = tile.getColumn() - this.currentCol;
+        // Calculate the row and column differences between the target tile and current tile
+        int rowDiff = targetTile.getRow() - this.currentTile.getRow();
+        int colDiff = targetTile.getColumn() - this.currentTile.getColumn();
 
         // Check if tile is occupied by another piece
-        if (tile.isOccupied()) {
+        if (targetTile.isOccupied()) {
             return false;
         }
 
         // If not a King
         if (!isKinged()) {
+
             // If not a King, check if the move is one space diagonally in the specified direction
             if (rowDiff == this.direction &&
-                    Math.abs(colDiff) == this.direction) {
+                    Math.abs(colDiff) == 1) {
                 return true;
+            } else if (rowDiff == this.direction * 2 && Math.abs(colDiff) == 2) {
 
-            } else if (rowDiff == 2 && Math.abs(colDiff) == 2){
-                // Calculate the position of the jumped tile
-                jumpedRow = (this.currentRow + rowDiff) / 2;
-                jumpedCol = (this.currentCol + colDiff) / 2;
+                // Check if there's a jump move and calculate the position of the jumped tile
+                jumpedRow = this.currentTile.getRow() + (rowDiff / 2);
+                jumpedCol = this.currentTile.getColumn() + (colDiff / 2);
+
+                // Removed the captured opponent's piece from the board
+                board[jumpedRow][jumpedCol] = null;
+
+                // Update the board with the new position of the moving piece
+                board[targetTile.getRow()][targetTile.getColumn()] =
+                        board[currentTile.getRow()][currentTile.getColumn()];
 
                 // Check if there is an opponent's piece on the jumped tile
-                if (tile.isOccupied() && tile.getOccupant().getPlayer() != this.player) {
-                    return true;
-                }
+                return targetTile.isOccupied() && canCapture(targetTile.getOccupant());
```

Comment:
Use jumpedRow and jumpedCol to get the jumpedTile from the board. Return if jumpedTile is occupied and if jumpedTile's occupant can be captured.

Reply:
Tile jumpedTile = board[jumpedRow][jumpedCol];